docs/reference/files/requirements-txt-file.rst

```diff
@@ -6,9 +6,13 @@
 
 The ``requirements.txt`` file is a standard Python `requirements file
 <https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format>`_
-used to declare and pin the version of any Python libraries required by a charm in
-production. This will be pre-populated with :literalref:`ops`. Any dependencies
-specified here will be bundled with the charm when it is built with
-:ref:`charmcraft pack <ref_commands_pack>`.
+used to declare and pin the version of any Python packages required by your charm.
+
+For :ref:`ref_commands_init` with the ``kubernetes`` or ``machine`` profile, Charmcraft
+doesn't create this file. Instead, Charmcraft creates a dependencies key in the
+:ref:`pyproject-toml-file`.
 
-.. _ops: https://ops.readthedocs.io/en/latest/
+For ``init`` with a 12-factor profile (``django-framework`` and so on), Charmcraft
```

Comment:
@javierdelapuente @alithethird I'd love it if we moved the 12-factor templates over to `uv` too, though we don't have to do it right now. WDYT?